### ISSUE TITLE
restricted-mode: pkgs.fetchurl vs builtins.fetchurl

### ISSUE THREAD

**Issue by dtzWill** [NixOS/nix#2043]
`builtins.fetchurl` is only allowed to access URI's in `allowed-uris`, but `pkgs.fetchurl` can access network in restricted mode "freely"?
Was just a bit surprised by this-- expected same behavior (when both are given a sha256).

Is this intended? What's the motivation for restricted-mode preventing fixed-output derivations? Thanks!

**Comment by edolstra** [comment]
Yes, this is intended. Restricted evaluation mode, as the name implies, only applies to evaluation.
